relayer/band/client.go

```diff
@@ -52,41 +54,61 @@ func NewQueryClient(
 
 // client is the BandChain client struct.
 type client struct {
-	Context      cosmosclient.Context
-	QueryClient  *QueryClient
-	Log          *zap.Logger
-	RpcEndpoints []string
+	Context     cosmosclient.Context
+	QueryClient *QueryClient
+	Log         *zap.Logger
+	Config      *Config
 }
 
 // NewClient creates a new BandChain client instance.
-func NewClient(ctx cosmosclient.Context, queryClient *QueryClient, log *zap.Logger, rpcEndpoints []string) Client {
+func NewClient(queryClient *QueryClient, log *zap.Logger, bandChainCfg *Config) Client {
+	encodingConfig := MakeEncodingConfig()
+	ctx := cosmosclient.Context{}.
+		WithCodec(encodingConfig.Marshaler).
+		WithInterfaceRegistry(encodingConfig.InterfaceRegistry)
+
 	return &client{
-		Context:      ctx,
-		QueryClient:  queryClient,
-		Log:          log,
-		RpcEndpoints: rpcEndpoints,
+		Context:     ctx,
+		QueryClient: queryClient,
+		Log:         log,
+		Config:      bandChainCfg,
 	}
 }
 
-// Connect connects to the BandChain using the provided RPC endpoints.
-func (c *client) Connect(timeout uint) error {
-	for _, rpcEndpoint := range c.RpcEndpoints {
+// Init initializes the BandChain client by connecting to the chain and starting
+// periodic liveliness checks.
+func (c *client) Init(ctx context.Context) error {
+	timeout := uint(c.Config.Timeout)
+	if err := c.connect(timeout, true); err != nil {
+		c.Log.Error("Failed to connect to BandChain", zap.Error(err))
+		return err
+	}
+
+	go c.startLivelinessCheck(ctx, timeout, c.Config.LivelinessCheckingInterval)
+	return nil
+}
+
+// connect connects to the BandChain using the provided RPC endpoints.
+func (c *client) connect(timeout uint, onStartup bool) error {
+	for _, rpcEndpoint := range c.Config.RpcEndpoints {
```

Comment:
Should we pass the context and use it? every function that call connect receives a ctx as a parameter?

Reply:
per discussed, aligned w/ go-relayer logic